src/middlewared/middlewared/plugins/smb.py

```diff
@@ -1143,6 +1238,28 @@ async def validate_aux_params(self, data, schema_name):
 
         verrors.check()
 
+    @private
+    async def cluster_share_validate(self, data, schema_name, verrors):
+        ha_mode = SMBHAMODE[(await self.middleware.call('smb.get_smb_ha_mode'))]
+        if ha_mode != SMBHAMODE.CLUSTERED:
+            return
+
+        if data['shadowcopy']:
+            verrors.add(
+                f'{schema_name}.shadowcopy',
+                'Shadow Copies are not implemented for clustered shares.'
+            )
+        if data['fsrvp']:
+            verrors.add(
+                f'{schema_name}.fsrvp',
+                'FSRVP support is not implemented for clustered shares.'
+            )
+        if not data['cluser_volname']:
```

Comment:
`KeyError` because of typo here

```suggestion
        if not data['cluster_volname']:
```